### ISSUE TITLE
Scaffolding Standard Identity Tables Fails to Produce AspNetUserRoles class

### ISSUE THREAD

**Issue by fsmirne** [npgsql/EntityFramework6.Npgsql#182]
I have a PostgreSQL database that was created using the .NET 3.1 identity framework.
I recently created a project using the .NET 6.01 libraries and tried to scaffold the tables, but I see that it's not creating the **AspNetUserRoles** table.

The table was created using the standard entity below:

```
modelBuilder.Entity("Microsoft.AspNetCore.Identity.IdentityUserRole<string>", b =>
{
    b.Property<string>("UserId")
        .HasColumnType("text");

    b.Property<string>("RoleId")
        .HasColumnType("text");

    b.HasKey("UserId", "RoleId");

    b.HasIndex("RoleId");

    b.ToTable("AspNetUserRoles");
});
```

Since this is a standard identity table, I don't see why it's not being produced by the scaffolding function. I believe there's a bug here. I first found out about this by using LINQPad and it was also failing to make this table available for querying.

Let me know if you need any more data to narrow what's happening and I'll be happy to provide.

**Comment by roji** [comment]
@fsmirne this is likely because EF Core 6.0 introduced support for recognizing many-to-many join tables (like AspNetUserRoles), and no longer produces a code entity for them by default; this is because EF Core's many-to-many support doesn't require an explicit join entity: if you create a database from the scaffolded model you should see that table getting created.

**Comment by fsmirne** [comment]
@roji this is interesting. If I create a web project in visual studio using .NET 6, the same code for creating the many to many table is present. In fact it's identical to the .NET 3.1 code. What do you think is the correct behavior here? It seems to me that creating the classes from the database should round trip correctly but scaffolding is not producing code that works for querying an existing table like that.

I'd like to hear your opinion on this.
Thank you so much.

**Comment by roji** [comment]
@fsmirne it's possible that the Visual Studio *template* for Identity includes the join entity explicitly; that's not incorrect, though it may be possible to remove it and get a simpler template. When creating a database from that template and then reverse-engineering it, it makes sense that EF Core wouldn't create the join entity since again, it's not strictly necessary.

Note that in general, EF Core doesn't really aim to provide full round-trippability when scaffolding a model and then creating a new database from it (or vice versa).

**Comment by fsmirne** [comment]
@roji I agree 100% with what you said. EF Core can automatically create the table without the join entity.
The only problem is ***if*** scaffolding doesn't create the class and specifically `public virtual DbSet<AspNetUserRole> AspNetUserRoles { get; set; }` then it's not possible to query that table directly, which seems like an incomplete job from the scaffolding function. I just verified that this behavior also happens with SQL Server, so I may open an issue there and reference this issue.

**Comment by roji** [comment]
> The only problem is if scaffolding doesn't create the class and specifically public virtual DbSet<AspNetUserRole> AspNetUserRoles { get; set; } then it's not possible to query that table directly, which seems like an incomplete job from the scaffolding function.

That's by-design - when the join table has no extra columns and so only associates the two sides (that seems like the right default for that case). You can always add an explicit entity type for the join table and do whatever you want; see some discussion [here](https://github.com/dotnet/efcore/issues/22475#issuecomment-921676340), we plan to allow tweaking this in the future.
